### ISSUE TITLE
GURA02 should not warn here.

### ISSUE THREAD

**Issue by JohanLarsson** [GuOrg/Gu.Roslyn.Asserts#123]
```cs
        [TestCase("GetProperty↓(\"Item\")")]
        [TestCase("GetProperty↓(\"Item\", BindingFlags.Public | BindingFlags.Instance)")]
        public static void IndexerAndPropertyNamedItem(string call)
        {
            var baseCode = @"
namespace RoslynSandbox
{
    public class Base
    {
        public int Item { get; }
    }
}";

            var code = @"
namespace RoslynSandbox
{
    using System.Reflection;

    public class C : Base
    {
        public C()
        {
            _ = typeof(C).GetProperty↓(""Item"");
        }

        public int this[int i] => 0;
    }
}".AssertReplace("GetProperty↓(\"Item\")", call);
            RoslynAssert.Diagnostics(Analyzer, ExpectedDiagnostic, baseCode, code);
        }
```
